File: src/main/java/com/manning/simplysend/repositories/OrdersRepository.java
package com.manning.simplysend.repositories;

import com.manning.simplysend.entities.Order;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface OrdersRepository extends PagingAndSortingRepository<Order, Long> {
    Page<Order> findAllByUser(String user, Pageable pageable);

    Page<Order> findAllByStatus(Order.StatusEnum status, Pageable pageable);

    Page<Order> findAllByUserAndStatus(String user, Order.StatusEnum status, Pageable pageable);
}

File: src/main/kotlin/com/manning/simplysend/repositories/OrdersRepository.kt
package com.manning.simplysend.repositories

import com.manning.simplysend.entities.Order
import com.manning.simplysend.entities.Order.StatusEnum
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository

@Repository
interface OrdersRepository : PagingAndSortingRepository<Order, Long> {

    fun findAllByUser(user: String, pageable: Pageable): Page<Order>

    fun findAllByStatus(status: StatusEnum, pageable: Pageable): Page<Order>

    fun findAllByUserAndStatus(user: String, status: StatusEnum, pageable: Pageable): Page<Order>
}